lib/whitespace.coffee

```diff
@@ -31,14 +31,26 @@ class Whitespace
         if atom.config.get('whitespace.ensureSingleTrailingNewline', scope: scopeDescriptor)
           @ensureSingleTrailingNewline(editor)
 
+    editorTextInsertedSubscription = editor.onDidInsertText (event) =>
+      return unless event.text is '\n'
+      scopeDescriptor = editor.getRootScopeDescriptor()
+
+      if atom.config.get('whitespace.removeTrailingWhitespace', scope: scopeDescriptor)
+        unless atom.config.get('whitespace.ignoreWhitespaceOnlyLines', scope: scopeDescriptor)
+          editor.transact =>
```

Comment:
@maxbrunsfeld Not particularly. I think I misunderstood what the transact method did. Now that you mention it, it actually isn't needed in this case. Removing seems to have no effect other than reducing the the steps before the a change is made.